src/GafferBindings/CompoundNumericPlugBinding.cpp

```diff
@@ -55,58 +55,59 @@ using namespace Gaffer;
 namespace
 {
 
-template<typename T>
-std::string compoundNumericPlugRepr( const T *plug )
+std::string maskedCompoundNumericPlugRepr( const ValuePlug *plug, unsigned flagsMask )
 {
-	return maskedCompoundNumericPlugRepr( plug, Plug::All );
-}
+	std::string extraArgs = "";
 
-template<typename T>
-class CompoundNumericPlugSerialiser : public ValuePlugSerialiser
-{
-  public:
-
-	static std::string repr( const Gaffer::ValuePlug *plug, unsigned flagsMask, const std::string &extraArguments, const Serialisation *serialisation )
+	if( const V3fPlug *v3fPlug = IECore::runTimeCast<const V3fPlug>( plug ) ) 
 	{
-		std::string result = ValuePlugSerialiser::repr( plug, flagsMask, extraArguments, serialisation );
-		if( const V3fPlug *v3fPlug = static_cast<const V3fPlug *>( plug ))
+		// TODO: we currently can't do this, but that would remove some of the uglyness below
```

Comment:
`IECorePython::repr()` is a misleading oddity that I don't really remember the reason for. It sounds like it should call into python and do `repr( o )`, but instead it's actually a bunch of overloads implemented in pure C++, that give you the equivalent for whatever types it's implemented for. I think either it was for performance or because at the time we didn't know how to actually do it properly, but either way I think we'd be better to remove it.

This is what I was thinking of doing, but hadn't explained sufficiently. It calls into Python to do the equivalent of `repr( plug.interpretation() ` :

```
boost::python::object interpretationAsPythonObject( plug->interpretation() );
boost::python::object interpretationRepr = i.attr( "__repr__" )();
std::string s = boost::python::extract<std::string>( r );
```

Unfortunately this reveals that `repr( interpretation )` isn't doing the right thing, and returns `IECore._IECore.Interpretation.Point` rather than `IECore.GeometricData.Interpretation.Point`. I think though that'd I'd rather still take this approach and replace the prefix with the right thing, so that we're robust to any addition enum values that might be added in cortex in the future.

We should also omit adding it to the extraArgs entirely when `interpretation == GeometricData::None` (which is the default anyway), so we get a more readable serialisation where possible.